src/test/components/Viewport.test.js

```diff
@@ -0,0 +1,603 @@
+      expect(getChartViewport().viewportLeft).toBeLessThan(viewportLeft);
+      expect(getChartViewport().viewportRight).toBeLessThan(viewportRight);
+
+      // Drag back the other way.
+      clickAndDrag(middleX, middleY, middleX - 1000, middleY);
+      expect(getChartViewport().viewportLeft).toBeGreaterThan(viewportLeft);
+      expect(getChartViewport().viewportRight).toBeGreaterThan(viewportRight);
+    });
+
+    it('will not scroll off to the left of the viewport bounds', function() {
+      const { scroll, getChartViewport, clickAndDrag } = setup();
+      scroll({
+        // Zoom in some large arbitrary amount:
+        deltaY: -5000,
+        shiftKey: true,
+        clientX: BOUNDING_BOX_LEFT + BOUNDING_BOX_WIDTH * 0.5,
+      });
+
+      // Perform the dragging action some arbitrarily large distance..
+      clickAndDrag(middleX, middleY, middleX + 10000, middleY);
+      expect(getChartViewport().viewportLeft).toBe(0);
+      expect(getChartViewport().viewportRight).toBeLessThan(0.4);
+    });
+
+    it('will not scroll off to the right of the viewport bounds', function() {
+      const { scroll, getChartViewport, clickAndDrag } = setup();
+
+      scroll({
+        // Zoom in some large arbitrary amount:
+        deltaY: -5000,
+        shiftKey: true,
+        clientX: BOUNDING_BOX_LEFT + BOUNDING_BOX_WIDTH * 0.5,
+      });
+
+      // Perform the dragging action some arbitrarily large distance.
+      clickAndDrag(middleX, middleY, middleX - 10000, middleY);
+      expect(getChartViewport().viewportLeft).toBeGreaterThan(0.6);
+      expect(getChartViewport().viewportRight).toBe(1);
+    });
+  });
+});
+
+function setup(profileOverrides: Object = {}) {
+  const flushRafCalls = mockRaf();
+  const ctx = mockCanvasContext();
+
+  jest
+    .spyOn(HTMLCanvasElement.prototype, 'getContext')
+    .mockImplementation(() => ctx);
+
+  jest
+    .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+    .mockImplementation(() => {
+      const rect = getBoundingBox(BOUNDING_BOX_WIDTH, BOUNDING_BOX_HEIGHT);
+      // Add some arbitrary offset to the bounding box to ensure that we
+      // are doing the correct thing when doing sizing calculations.
+      rect.left += BOUNDING_BOX_LEFT;
+      rect.right += BOUNDING_BOX_LEFT;
+      rect.x += BOUNDING_BOX_LEFT;
+      rect.y += BOUDNING_BOX_TOP;
+      rect.top += BOUDNING_BOX_TOP;
+      rect.bottom += BOUDNING_BOX_TOP;
+      return rect;
+    });
+
+  // Hook up a dummy chart with a viewport.
+  const DummyChart = () => <div id="dummy-chart" />;
+  const ChartWithViewport = withChartViewport(DummyChart);
+
+  // The viewport component started out as an unconnected component, but then it
+  // started subscribing to the store an dispatching its own actions. This migration
+  // wasn't completely done, so it still has a few pieces of state passed in through
+  // its OwnProps. In order to ensure that the component is consistent, make it
+  // a connected component.
```

Comment:
@gregtatum I still dont understand why we need to explicitely connect it in the tests :) We don't do this in the real code ever, and all this does is providing a `timeRange` and `previewSelection` that never change (so that you could provide directly).

The component returned by `withChartViewport` is already connected, and _yes_ we need to provide it the `store` like you're properly doing below. But I think that the extra explicit connection done here is 1/ useless and 2/ not like the actual code using this HOC.